modules/volunteers/admin/sitesupport/checkout.js

```diff
@@ -0,0 +1,223 @@
+/* jshint esversion: 6 */
+/* globals  apiRequest, showSidebar, hideSidebar, confirmbox, alertbox, showSpinner, hideSpinner */
+
+export default {
+  data() {
+    return {
+      checkout: [],
+      hoursSpent: 0,
+      groupsNow: [],
+      failedList: null,
+    }
+  },
+  methods: {
+    processCheckout() {
+      confirmbox('Confirm Distribute Gifts',
+        'Are the selected gifts correct?').then(() => {
+        showSpinner();
+        var prizes = [this.checkout.length, 0, 0];
+        this.checkout.forEach((item) => {
+          console.log(item);
+          for (let n = 0; n < item.count; n++) {
```

Comment:
How high of a count can we expect for items? If we have 50 items, this could potentially send 50 simultaneous requests and these will execute as promises and are non-blocking. If we want to block and wait for each one, or even limit to a chunk of requests at a time we may want to consider a different approach.

Reply:
I would be surprised at 50, I would say at most 10.  We sort of block and wait as we have a please wait spinner going until all the request have succeeded or failed.  Then we report the result.  I do not feel this is terrible (but correct me if it make your web developer skin crawl)  we could modify the API endpoint to accept a list and then report results for each item.  Then it would be 1 request.

Reply:
Depends on the implementation - my concern is mostly, do we care if 10 requests are fire simultaneously? It is only 1 user click, but 10 requests all at once. 

Sending the full list of items and processing the full list would be ideal here if we can do that.